block/manager.go

```diff
@@ -101,9 +101,27 @@ type Manager struct {
 
 // getInitialState tries to load lastState from Store, and if it's not available it reads GenesisDoc.
 func getInitialState(store store.Store, genesis *cmtypes.GenesisDoc) (types.State, error) {
+	// Load the state from store.
 	s, err := store.GetState()
 	if err != nil {
+		// If the user is starting a fresh chain (or hard-forking), we assume the stored state is empty.
 		s, err = types.NewFromGenesisDoc(genesis)
```

Comment:
i think we should also call `store.SetHeight(s.LastBlockHeight)` after `NewFromGenesisDoc`, otherwise the publishBlock will try to `GetCommit` with `m.store.Height()` which is 0 (https://github.com/rollkit/rollkit/blob/main/block/manager.go#L667), we want to proceed like genesis with empty last commit: https://github.com/rollkit/rollkit/blob/main/block/manager.go#L665

Reply:
Agree